flink-core/src/main/java/org/apache/flink/api/common/io/GenericCsvInputFormat.java

```diff
@@ -300,14 +300,14 @@ protected void setFieldsGeneric(boolean[] includedMask, Class<?>[] fieldTypes) {
 	// --------------------------------------------------------------------------------------------
 	//  Runtime methods
 	// --------------------------------------------------------------------------------------------
-	
+
 	@Override
-	public void open(FileInputSplit split) throws IOException {
-		super.open(split);
+	public void initializeSplit(FileInputSplit split, Long offset) throws IOException {
+		super.initializeSplit(split, offset);
 
 		// instantiate the parsers
 		FieldParser<?>[] parsers = new FieldParser<?>[fieldTypes.length];
```

Comment:
Move this logic to the open method? It is not really related to the split initialization itself.

Reply:
we would have to move it to both open and reopen now this is why I introduced a common init method that is called in both instead.

Better would be moving to configure but unfortunately configure is not called consistently across the codebase and did not want to open another pandora's box

Reply:
Sorry, I was too quick with this comment. I didn't check whether the parsers are stateful. In this case, we need to recreate them during `reopen()`.